generator/src/main/java/com/scottlogic/deg/generator/inputs/ProfileReader.java

```diff
@@ -17,11 +19,20 @@
 import java.util.stream.Collectors;
 
 public class ProfileReader {
-    public Profile read(Path filePath) throws IOException, InvalidProfileException {
+    public Profile read(Path filePath, boolean doProfileValidation) throws IOException, InvalidProfileException {
         byte[] encoded = Files.readAllBytes(filePath);
         String profileJson = new String(encoded, Charset.forName("UTF-8"));
 
-        return this.read(profileJson);
+        Profile profile =  this.read(profileJson);
+
+        if( doProfileValidation) {
+            ProfileValidationVisitor profileValidationVisitor = new ProfileValidationVisitor();
+            profile.accept(profileValidationVisitor);
+            CmdProfileValidationReporter profileValidationReporter = new CmdProfileValidationReporter();
```

Comment:
1. Should be a dependency
1. Don't mention `cmd` this implies a Windows subsystem, the tool should be agnostic to the environment it runs on - maybe "SystemOut" or "StandardOut".

Reply:
done